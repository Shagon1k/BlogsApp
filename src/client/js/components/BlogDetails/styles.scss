.blogDetails {
	width: 500px;
	height: 300px;
	margin: auto auto;
}