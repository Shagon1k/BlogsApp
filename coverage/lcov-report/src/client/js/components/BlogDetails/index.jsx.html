<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/client/js/components/BlogDetails/index.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\..\..\index.html">All files</a> / <a href="index.html">src/client/js/components/BlogDetails</a> index.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">80% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>8/10</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">75% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>3/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>6/6</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27</td><td class="line-coverage quiet"><span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import { Link } from 'react-router-dom';
import './styles.scss';
&nbsp;
const BlogDetails = ({blog}) =&gt; {
	if (blog) 
		return(
			&lt;div className="blogDetails"&gt;
				&lt;dl&gt;
					&lt;dt&gt; Title: &lt;/dt&gt;
					&lt;dd&gt; {blog.title} &lt;/dd&gt;
					&lt;dt&gt; Author: &lt;/dt&gt;
					&lt;dd&gt; {blog.author} &lt;/dd&gt;
					&lt;dt&gt; Message: &lt;/dt&gt;
					&lt;dd&gt; {blog.message}&lt;/dd&gt;
					&lt;dt&gt; Date: &lt;/dt&gt;
					&lt;dd&gt; {(new Date(blog.date)).toLocaleString()}&lt;/dd&gt;
				&lt;/dl&gt;
				&lt;Link to='/blogs'&gt;Blogs main&lt;/Link&gt;
			&lt;/div&gt;
		)
	else return(
		&lt;div&gt;No such blog&lt;/div&gt;
		)
}
&nbsp;
export default BlogDetails;<span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="missing-if-branch" title="else path not taken" >E</span></span></span></pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Mar 02 2018 16:38:36 GMT+0300 (Belarus Standard Time)
</div>
</div>
<script src="..\..\..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\..\..\sorter.js"></script>
</body>
</html>
