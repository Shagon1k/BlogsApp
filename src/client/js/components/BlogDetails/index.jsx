import React from 'react';
import { Link } from 'react-router-dom';
import './styles.scss';

const BlogDetails = ({blog}) => {
	if (blog) 
		return(
			<div className="blogDetails">
				<dl>
					<dt> Title: </dt>
					<dd> {blog.title} </dd>
					<dt> Author: </dt>
					<dd> {blog.author} </dd>
					<dt> Message: </dt>
					<dd> {blog.message}</dd>
					<dt> Date: </dt>
					<dd> {(new Date(blog.date)).toLocaleString()}</dd>
				</dl>
				<Link to='/blogs'>Blogs main</Link>
			</div>
		)
	else return(
		<div>No such blog</div>
		)
}

export default BlogDetails;