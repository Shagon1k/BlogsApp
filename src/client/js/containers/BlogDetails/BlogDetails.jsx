import { connect } from 'react-redux';
import BlogDetails from '../../components/BlogDetails/index.jsx';

const mapStateToProps = (state, ownProps) => {
	const blogId = ownProps.match.params.id;
	const blogObj = state.blogs.find((i) => i.id === blogId)
	return {
		blog: blogObj ? blogObj.blog : null
	}
}

const BlogDetailsContainer = connect(
	mapStateToProps,
	null
)(BlogDetails)

export default BlogDetailsContainer;