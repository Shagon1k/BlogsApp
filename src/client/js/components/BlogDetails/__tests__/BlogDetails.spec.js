import React from 'react';
import BlogDetails from '../index.jsx';

describe('BlogDetails component', () => {
	it('renders blogs details correctly', () => {
	const blog = {
		title: 'title',
		author: 'author',
		message: 'message',
		date: '2018-03-01T13:37:08.068Z'
	};
	const wrapper = shallow(<BlogDetails blog={blog}/>)
	expect(wrapper).toMatchSnapshot();
	});
	
	it('renders no blog info correctly', () => {
		const blog = null;
		const wrapper = shallow(<BlogDetails blog={blog} />);
		expect(wrapper).toMatchSnapshot();
	})
})
