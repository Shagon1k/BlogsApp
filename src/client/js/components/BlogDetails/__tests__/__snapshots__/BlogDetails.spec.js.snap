// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogDetails component renders blogs details correctly 1`] = `
<div
  className="blogDetails"
>
  <dl>
    <dt>
       Title: 
    </dt>
    <dd>
       
      title
       
    </dd>
    <dt>
       Author: 
    </dt>
    <dd>
       
      author
       
    </dd>
    <dt>
       Message: 
    </dt>
    <dd>
       
      message
    </dd>
    <dt>
       Date: 
    </dt>
    <dd>
       
      3/1/2018, 4:37:08 PM
    </dd>
  </dl>
  <Link
    replace={false}
    to="/blogs"
  >
    Blogs main
  </Link>
</div>
`;

exports[`BlogDetails component renders no blog info correctly 1`] = `
<div>
  No such blog
</div>
`;
